### PR TITLE
bpo-31862: Port binascii to PEP 489 multiphase initialization

### PR DESCRIPTION
**Pull request by ghost**
This PR ports `_bisect` to the PEP 489 multiphase initialization.

The module doesn't use mutable global state, so it should be OK to use with subinterpreters.

This PR is deliberately simple to test the workflow; more and more complicated ones will follow.

https://bugs.python.org/issue31862

### CODE REVIEW

File: Misc/NEWS.d/next/Core and Builtins/2017-10-24-17-26-58.bpo-31862.5Gea8L.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Port md5 module to PEP 489 multiphase initialization.
```
Comment (merwok) [inline Misc/NEWS.d/next/Core and Builtins/2017-10-24-17-26-58.bpo-31862.5Gea8L.rst]: Do you mean _bisect module?

Comment (ghost) [reply Misc/NEWS.d/next/Core and Builtins/2017-10-24-17-26-58.bpo-31862.5Gea8L.rst]: Yes, sorry for that.

**Comment by ghost** [comment]
What is the best solution to porting of this many modules?
Should I add all of the modules to this PR, or is it better to merge ported modules in groups in multiple PR's?

**Comment by ghost** [comment]
@ncoghlan Would you mind reviewing this?

**Review by auvipy — APPROVED** [review]

**Comment by ncoghlan** [comment]
Hi @Traceur759 - sorry for leaving this unreviewed for so long. Now that @ericsnowcurrently's draft internal API for Python-level subinterpreter support has been added, would you be able to come up with a test case for these modules that fails without the patch, and works with it?

It would also be nice to avoid the module state retrieval overhead in the cases where an exception *doesn't* need to be raised - I know you have a draft PEP for defining static shared exception types, for this PR specifically you could avoid the problem by deferring loading the state until an exception is about to be raised.

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](https://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for your contribution, we look forward to reviewing it!

**Comment by brettcannon** [comment]
Looks like @Dormouse759 doesn't have a bugs.python.org account associated with that GitHub username, so closing as the CLA isn't signed still. This PR can be re-opened if that changes.

**Comment by Dormouse759** [comment]
@brettcannon I have renamed my account and forgotten to change the GH name on bugs.python.org.
It should be associated now.

**Comment by brettcannon** [comment]
@Dormouse759 thanks for the fix! I've reopened the PEP.

**Comment by encukou** [comment]
@Dormouse759 I rebased on current master, and I have one more suggestion, which is better expressed as a commit rather than text :)
Do you agree with this extra change in `binascii_exec`?

**Comment by Dormouse759** [comment]
@encukou Yes, it seems fine. Thank you.

**Pull request merged**